app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt

```diff
@@ -0,0 +1,350 @@
+    activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+    onDispose { activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED }
+  }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Modify your profile information") },
+            navigationIcon = { GoBackButton(navigationActions) },
+            colors = TopAppBarDefaults.topAppBarColors(containerColor = colorScheme.background))
+      },
+      content = { padding ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(padding)
+                    .background(colorScheme.background)
+                    .padding(16.dp)
+                    .verticalScroll(rememberScrollState()),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Top) {
+              val user = authViewModel.user.collectAsState()
+              val userId = user.value?.uid ?: "-1"
+              val provider =
+                  listProviderViewModel.providersList.collectAsState().value.find {
+                    it.uid == userId
+                  } ?: return@Column
+
+              ModifyInput(provider = provider, navigationActions = navigationActions)
+            }
+      })
+}
+
+@Composable
+fun ModifyInput(
+    provider: Provider,
+    locationViewModel: LocationViewModel = viewModel(factory = LocationViewModel.Factory),
+    listProviderViewModel: ListProviderViewModel =
+        viewModel(factory = ListProviderViewModel.Factory),
+    navigationActions: NavigationActions
+) {
+  val context = LocalContext.current
+  var newCompanyName by remember { mutableStateOf(provider.companyName) }
+  val okNewCompanyName = newCompanyName.length >= 2 && newCompanyName.isNotBlank()
+  var newProfession by remember { mutableStateOf(provider.service) }
+
+  var newPhoneNumber by remember { mutableStateOf(provider.phone) }
+  val okNewPhoneNumber =
+      newPhoneNumber.isNotBlank() &&
+          newPhoneNumber.all { it.isDigit() || it == '+' } &&
+          newPhoneNumber.length > 6
+
+  val newLocation by remember { mutableStateOf(provider.location.name) }
+
+  var showDropdown by remember { mutableStateOf(false) }
+
+  val locationSuggestions by
+      locationViewModel.locationSuggestions.collectAsState(initial = emptyList<Location?>())
+  var selectedLocation by remember { mutableStateOf<Location?>(provider.location) }
+  val okNewLocation = selectedLocation != null
+
+  var newLanguage by remember { mutableStateOf(provider.languages) }
+
+  val allIsGood = okNewCompanyName && okNewPhoneNumber && okNewLocation
+
+  CustomOutlinedTextField(
+      value = newCompanyName,
+      onValueChange = { newCompanyName = it },
+      label = "Provider Name",
+      placeholder = "Enter your new company name",
+      isValueOk = okNewCompanyName,
+      leadingIcon = Icons.Default.AccountCircle,
```

Comment:
You should put company name instead of provider name, otherwise it's a bit confusing for the user. And maybe we should also add a field to edit the name?

Reply:
Is was provider name. I did a mistake. It is fix now!